tests/phpunit/tests/rest-api/class-wp-rest-block-directory-controller-test.php

```diff
@@ -0,0 +1,194 @@
+		$request->set_query_params( array( 'term' => 'foo' ) );
+
+		$this->prevent_requests_to_host( 'api.wordpress.org' );
+
+		$result = @ $this->controller->get_items( $request );
+		$this->assertWPError( $result );
+		$this->assertTrue( array_key_exists( 'plugins_api_failed', $result->errors ), 'Returns the correct error key' );
+
+	}
+
+	/**
+	 * Tests that the install endpoint returns WP_Error when the server is unreachable.
+	 */
+	function test_install_unreachable() {
+		$request = new WP_REST_Request( 'POST', '/wp/v2/block-directory/install' );
+		$request->set_query_params( array( 'slug' => 'foo' ) );
+
+		$this->prevent_requests_to_host( 'api.wordpress.org' );
+
+		$result = @ $this->controller->install_block( $request );
+		$this->assertWPError( $result );
+		$this->assertTrue( array_key_exists( 'plugins_api_failed', $result->errors ), 'Returns the correct error key' );
+
+	}
+
+	/**
+	 * Should fail with a permission error if requesting user is not logged in.
+	 */
+	function test_simple_search_no_perms() {
+		$request  = new WP_REST_Request( 'GET', '/wp/v2/block-directory/search' );
+		$request->set_query_params( array( 'term' => 'foo' ) );
+		$response = rest_get_server()->dispatch( $request );
+		$data     = $response->get_data();
+
+		$this->assertEquals( $data['code'], 'rest_user_cannot_view' );
+	}
+
+	/**
+	 * Make sure a search with the right permissions returns something.
+	 */
+	function test_simple_search_with_perms() {
+		$user_id = $this->factory->user->create( array( 'role' => 'administrator' ) );
+		wp_set_current_user( $user_id );
+
+		// This will hit the live API. We're searching for `block` which should definitely return at least one result.
+		$request  = new WP_REST_Request( 'GET', '/wp/v2/block-directory/search' );
+		$request->set_query_params( array( 'term' => 'block' ) );
+		$response = rest_get_server()->dispatch( $request );
+		$data     = $response->get_data();
+
+		$this->assertEquals( 200, $response->status );
+		// At least one result
+		$this->assertGreaterThanOrEqual( 1, count( $data ) );
+		// Each result should be an object with important attributes set
+		foreach ( $data as $plugin ) {
+			$this->assertArrayHasKey( 'name', $plugin );
+			$this->assertArrayHasKey( 'title', $plugin );
+			$this->assertArrayHasKey( 'id', $plugin );
+			$this->assertArrayHasKey( 'author_block_rating', $plugin );
+			$this->assertArrayHasKey( 'assets', $plugin );
+			$this->assertArrayHasKey( 'humanized_updated', $plugin );
+		}
+	}
+
+	/**
+	 * Should fail with a permission error if requesting user is not logged in.
+	 */
+	function test_simple_install_no_perms() {
+		$request  = new WP_REST_Request( 'POST', '/wp/v2/block-directory/install' );
+		$request->set_query_params( array( 'slug' => 'foo' ) );
+		$response = rest_get_server()->dispatch( $request );
+		$data     = $response->get_data();
+
+		$this->assertEquals( $data['code'], 'rest_user_cannot_view' );
```

Comment:
This will be able to use `assertErrorResponse` once we change the base class.

Reply:
Switched to assertErrorResponse in 511bbdb054.